### ISSUE TITLE
could this improve rendering latency for web browsers ? (2D GUIs)

### ISSUE THREAD

**Issue by LifeIsStrange** [ishitatsuyuki/LatencyFleX#1]

**Comment by ishitatsuyuki** [comment]
Unfortunately, I think there would be a lot of obstacles applying this to 2D GUIs. Here are some reasons:
- Web browser workloads are highly uneven. Sometimes they don’t update at all, sometimes they do a partial update, once in a while they do a full repaint. It’s drastically different from video games where workload changes smoothly frame-to-frame.
- 2D GUIs relies on VSync, and VSync heavily penalizes deadline misses. LatencyFleX currently assumes that missing the target time is OK, given that most games are ran inside FreeSync range or VSync off. I still haven’t decided what’s a good strategy to handle such an asymmetric deadline model.

**Comment by LifeIsStrange** [comment]
@kvark @nical @emilio friendly ping, I wonder wether you have thoughts to share on this topic

**Comment by nical** [comment]
I agree with ishitatsuyuki's assesment regarding how uneven browser workloads are and how vsync affects these things. Note that Firefox already does some things like refresh driver throttling when the end of the pipeline can't keep up with the start, although it's more to avoid catastrophic accumulation of gpu work than to fine tune latency.

**Comment by LifeIsStrange** [comment]
However, if Firefox could detect adaptative sync support that would mitigate the vsync issue no?

**Comment by nical** [comment]
In theory maybe, but you still have the problem of very uneven workloads. More importantly we already have some lower hanging (and more impactful) fruits to pick in terms of latency improvement including work in this year's roadmap. And variable refresh rate monitors are too uncommon as of today in proportion for us to dedicate more time than making sure it works as well as vsync.

**Comment by nonetrix** [comment]
If you really want lower latency for whatever reason in a web browser, you can always disable vsync in most browsers though flags or about:config to my understanding. I don't really know why it would be a issue in the context of a browser though, but browsers are so easy to run that you can get like 500+ FPS easily
